coordinator/internal/logic/provertask/batch_prover_task.go

```diff
@@ -311,9 +311,12 @@ func (bp *BatchProverTask) getBatchTaskDetail(dbBatch *orm.Batch, chunkProofs []
 	if !bp.validiumMode() {
 		dbBatchCodecVersion := encoding.CodecVersion(dbBatch.CodecVersion)
 		switch dbBatchCodecVersion {
+		case 0:
+			log.Warn("the codec version is 0, if it is not under integration test we have encountered an error here")
```

Comment:
Does this mean we use codec=0 for integration tests?

Reply:
In integration tests we only provided a dummy task with many fields unfilled, so the version field is 0.

And in integration test, the blob bytes are empty and can not be parsed by any codec, we need to return early here.

It is not a trivial task to refine the whole integration tests so currently we would better just keep this workaround for compatibility.